src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBResultSet.java

```diff
@@ -23,228 +23,253 @@
-    {
-        prefetch(n);
-        return nodeRefs[n];
-    }
-
-    /* (non-Javadoc)
-     * @see org.alfresco.service.cmr.search.ResultSetSPI#getRow(int)
-     */
-    @Override
-    public ResultSetRow getRow(int i)
-    {
-        return new DBResultSetRow(this, i);
-    }
-
-    /* (non-Javadoc)
-     * @see org.alfresco.service.cmr.search.ResultSetSPI#getChildAssocRef(int)
-     */
-    @Override
-    public ChildAssociationRef getChildAssocRef(int n)
-    {
-        ChildAssociationRef primaryParentAssoc = nodeService.getPrimaryParent(getNodeRef(n));
-        if(primaryParentAssoc != null)
-        {
-            return primaryParentAssoc;
-        }
-        else
-        {
-            return null;
-        }
-    }
-
-    /* (non-Javadoc)
-     * @see org.alfresco.service.cmr.search.ResultSetSPI#getResultSetMetaData()
-     */
-    @Override
-    public ResultSetMetaData getResultSetMetaData()
-    {
-        return resultSetMetaData;
-    }
-
-    /* (non-Javadoc)
-     * @see org.alfresco.service.cmr.search.ResultSetSPI#getStart()
-     */
-    @Override
-    public int getStart()
-    {
-        return 0;
-    }
-
-    /* (non-Javadoc)
-     * @see org.alfresco.service.cmr.search.ResultSetSPI#hasMore()
-     */
-    @Override
-    public boolean hasMore()
-    {
-        return false; 
-    }
-
-    /* (non-Javadoc)
-     * @see java.lang.Iterable#iterator()
-     */
-    @Override
-    public Iterator<ResultSetRow> iterator()
-    {
-        return new DBResultSetRowIterator(this);
-    }
-    
-    private void prefetch(int n)
-    {
-       
-        if (prefetch.get(n))
-        {
-            // The document was already processed
-            return;
-        }
-        // Start at 'n' and process the the next bulk set
-        int bulkFetchSize = getBulkFetchSize();
+import org.alfresco.util.Pair;
+
+/**
+ * @author Andy
+ *
+ */
+public class DBResultSet extends AbstractResultSet
+{
+    private List<Long> dbids;
+
+    private NodeRef[] nodeRefs;
+
+    private NodeDAO nodeDao;
+
+    private NodeService nodeService;
+
+    private TenantService tenantService;
+
+    private SimpleResultSetMetaData resultSetMetaData;
+
+    private BitSet prefetch;
+
+    private int numberFound;
+
+    public DBResultSet(SearchParameters searchParameters, List<Long> dbids, NodeDAO nodeDao, NodeService nodeService,
+            TenantService tenantService, int maximumResultsFromUnlimitedQuery)
+    {
+        this.nodeDao = nodeDao;
+        this.dbids = dbids;
+        this.nodeService = nodeService;
+        this.tenantService = tenantService;
+        this.prefetch = new BitSet(dbids.size());
+        nodeRefs = new NodeRef[(dbids.size())];
+        this.numberFound = dbids.size();
+
+        final LimitBy limitBy;
+        int maxResults = -1;
+        if (searchParameters.getMaxItems() >= 0)
+        {
+            maxResults = searchParameters.getMaxItems();
+            limitBy = LimitBy.FINAL_SIZE;
+        } else
```

Comment:
ACS code formatting requires all braces to be on new lines (except for lambda expressions)
See https://hub.alfresco.com/t5/alfresco-content-services-hub/coding-standards-for-alfresco-content-services/ba-p/290457#toc-hId-1652896845